server/util/util.ts

```diff
@@ -3,7 +3,7 @@ import moment from 'moment';
 import { BadRequestError } from './error-types';
 
 
-export function dateFromString(dateStr: string) {
+export function dateFromString(dateStr: string, shouldThrow = true) {
```

Comment:
Yeah, I definitely didn't like this approach. The problem is due to the nested function calls and where this exception would bubble up. As it was, the code throws a BadRequestError() - which in my opinion is too opinionated for such a utility type of function. I didn't want to go on a refactoring spree... maybe we should talk about this on a call.

Reply:
Agreed. I'd like to see this function possibly removed in favor of using a package like moment to handle it, and let the caller handle invalid input instead of manipulating it internally within the function.

We can just add this to the tech debt list since it will involve some refactoring.